docs/kubebuilder_annotation.md

```diff
@@ -0,0 +1,75 @@
+
+## Kubebuilder Annotation Syntax
+
+Kubebuilder Annotation has a series of tokens separated by colons into groups from left to right. Each **Token** is a string identifier in an annotation instance. It has meaning by its position in token slice, in the form of
+**+[header]:[module]:[submodule]:[key-value elements]**
+Go Annotation starts with `+` (e.g. `// +kubebuilder`) to differentiate from regular go comments.
+
+## Token types
+
+- **header** is the identifier of a group of annotations. It helps user know which project provides this annotation. For example, in Kubernetes project, headers like `kubebuilder`, `k8s`, `genclient`, etc. are all project identifiers. A header is required for all annotations, since you may use multiple annotations from different projects in the same codebase.
+
+- **module** is the identifier of functional module in an annotation. An annotation may have a group of modules, each of which performs a particular function.
+
+- **submodule** (optional) In some cases, the module has a big functional scope, split into fine-grained sub modules, which provide the flexibility of extending module functionality. For example: **module:submodule1:submodule2:submodule3** submodule can be multiple following one by one.
+
+## Levels of symbols
+
+Delimiter symbols are distinguished to work in different levels from top-down for splitting values string in tokens, which provides readability and efficiency.
+
+- **Colon**
+
+  Colon `:` is the 1st level delimiter (to annotation) only for separate tokens. Tokens on different sides of the colon should refer to different token types.
+
+- **Comma**
+
+  Comma `,` is the 2nd level delimiter (to annotation) for splitting key-value pairs in **key-value elements** which is normally the last token in annotation. e.g. `+kubebuilder:printcolumn:name=<name>,type=<type>,description=<desc>,JSONPath:<.spec.Name>,priority=<int32>,format=<format>` It works within token which is the 2nd level of annotation, so it is called "2nd level delimiter"
+
+- **Equal sign**
+
+  Equal sign `=` is the 3rd level delimiter (to annotation) for identifying key and value. Since the `key=value` parts are splitted from single token (2nd level), its inner delimiter `=` works for next level (3rd level)
+
+- **Pipe sign or Vertical bar**
+
+  Pipe sign `|` is the 4th level delimiter, which works inside the `key=value` part (3rd level) indicating key and value.
+
```

Comment:
We use `;` also as delimiter ? I remember seeing it somewhere. Also pl. check with core k8s convention.

Reply:
Right, `semicolon` is one of the delimiter. I missed this section in editing MD file. Added back.